### ISSUE TITLE
Publish App in F-Droid

### ISSUE THREAD

**Issue by M4GNV5** [neuland-ingolstadt/neuland.app-native#23]
https://gitlab.com/fdroid/fdroiddata/blob/master/CONTRIBUTING.md

**Comment by Robert27** [comment]
Yes, you're right. I completely overlooked transferring the issue. Sorry for that. We should definitely continue to follow the F-Droid release, but unfortunately I have no idea how. We look forward for your help, Chris!

**Comment by BuildmodeOne** [comment]
+1

**Comment by Robert27** [comment]
Currently not possible due to:
https://github.com/maplibre/maplibre-react-native/issues/433

**Comment by Robert27** [comment]
Also: “Please note that some [Expo](https://expo.io/) modules depend on non-free components, so they may be incompatible with F-Droid.”

https://f-droid.org/2020/10/14/adding-react-native-app-to-f-droid.html

**Comment by H3rmt** [comment]
> Currently not possible due to: [maplibre/maplibre-react-native#433](https://github.com/maplibre/maplibre-react-native/issues/433)

https://github.com/maplibre/maplibre-react-native/issues/433#issuecomment-2591181365

**Comment by Robert27** [comment]
We should look into this after upgrading to Expo SDK 53 / RN 0.70.

**Comment by Robert27** [comment]
@CrsiX please let me know if you need anything to complete this, like screenshots or legal advice etc

**Comment by CrsiX** [comment]
F-Droid builds do not support secrets whatsoever to make reproducible builds work fine. Therefore, it is not possible to add the API key for the THI API to the app during the build process, as the build process does not have any mechanism for "secrets". There are various ways to work around this, but they are all ugly and only listed for completeness.

- Just include the API key in the source code, this makes it effectively public for everybody. This got rejected for obvious reasons, so long as the API key must remain "secret enough" (i.e., it is included in the Google Play builds but could be extracted from them through decompiling etc. as well).
- Do not use any API key (authentication is still required additionally) - this requires changes to the THI API, which are not done yet and are unlikely to come.
- Find a way to hack around the limitations, e.g. let a build script fetch the API key from a server that only allows the GitLab/F-Droid build server IP addresses, or encrypt/obfuscate the API key in some way that allows the usage through the app. While the first option now requires us to maintain this (trusted) infrastructure and figure out all these details, the second option is by design extremely weak.
- Ask the user for the API key on first app usage after installation. This obviously is extremely bad user experience.

Did I miss something, @Robert27?

**Comment by McBaumwolle** [comment]
Have you considered publishing releases as APKs on GitHub? This would allow users to download the app directly or access it privacy friendly without Play via [Obtanium](https://github.com/ImranR98/Obtainium) (which works great in my experience).

**Comment by Robert27** [comment]
Thanks for your suggestion. The reason why we do not include the APK in the GitHub releases yet is that the extraction of the THI API key would be even easier. However I understand all the objections here and will discuss this with the board at the next organizational meeting.
